### PR TITLE
bump to go1.19

### PR DESCRIPTION
**Pull request by onyxraven**
Background
-----

Bump to latest things.

1. to run golangci-lint in 1.19, had to upgrade past 1.48 - 
2. this introduced some new lints

    ```
    oswrap/exec.go:16: File is not `goimports`-ed (goimports)
    //todo use oswrap.Instance() instead of package local ow?
    oswrap/os.go:4:2: SA1019: "io/ioutil" has been deprecated since Go 1.16: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details. (staticcheck)
    ```

Versioning
-----

Patch release

### CODE REVIEW

File: oswrap/exec.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ type ExecWrap interface {
 
 type execWrap struct{}
 
-//todo use oswrap.Instance() instead of package local ow?
+// todo use oswrap.Instance() instead of package local ow?
```
Comment (onyxraven) [inline oswrap/exec.go:16]: linter complaint ;)

File: oswrap/exec_test.go [hunk 2]
```diff
@@ -85,9 +84,8 @@ func TestRunCommandStdoutToFile(t *testing.T) {
 
 		mock.EXPECT().Create(gomock.Eq("filename")).DoAndReturn(func(f string) (*os.File, error) {
 			//TODO replace all file stuff with afero
-			return ioutil.TempFile("/tmp", "TestRunCommandStdoutToFile")
+			return os.CreateTemp(t.TempDir(), "TestRunCommandStdoutToFile")
 		})
-		defer os.Remove("/tmp/TestRunCommandStdoutToFile")
```
Comment (onyxraven) [inline oswrap/exec_test.go:90]: adding to the t.TempDir means it gets cleaned up by the test framework. No more weird remove (which I'm unsure actually worked)

File: oswrap/os.go [hunk 3]
```diff
@@ -51,8 +50,8 @@ func (ow osWrap) Stat(name string) (os.FileInfo, error) {
 	return os.Stat(name)
 }
 func (ow osWrap) ReadFile(filename string) ([]byte, error) {
-	return ioutil.ReadFile(filename)
+	return os.ReadFile(filename)
 }
 func (ow osWrap) WriteFile(filename string, data []byte, perm os.FileMode) error {
-	return ioutil.WriteFile(filename, data, perm)
+	return os.WriteFile(filename, data, perm)
```
Comment (onyxraven) [inline oswrap/os.go:56]: nice that these just worked.

**Review by kr3cj — DISMISSED** [review]

File: go.mod [hunk 4]
```diff
@@ -5,14 +5,13 @@ go 1.17
 require (
 	github.com/golang/mock v1.6.0
 	github.com/mozilla-services/yaml v0.0.0-20201007153854-c369669a6625
-	github.com/spf13/cobra v1.2.1
```
Comment (onyxraven) [inline go.mod:8]: this update reduced the dependency list for this package so much! yay!

File: go.mod [hunk 5]
```diff
@@ -5,14 +5,13 @@ go 1.17
...
 require (
-	github.com/cpuguy83/go-md2man/v2 v2.0.0 // indirect
-	github.com/inconshreveable/mousetrap v1.0.0 // indirect
-	github.com/russross/blackfriday/v2 v2.0.1 // indirect
-	github.com/shurcooL/sanitized_anchor_name v1.0.0 // indirect
+	github.com/cpuguy83/go-md2man/v2 v2.0.2 // indirect
+	github.com/inconshreveable/mousetrap v1.0.1 // indirect
+	github.com/russross/blackfriday/v2 v2.1.0 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
-	gopkg.in/yaml.v2 v2.4.0 // indirect
```
Comment (onyxraven) [inline go.mod:17]: we don't use this one, we use the mozilla one above. this is an indirect/transient from cobra

**Comment by onyxraven** [comment]
tested by hand too.

**Pull request merged**